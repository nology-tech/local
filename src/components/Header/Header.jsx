import AppLogo from "../../assets/images/AppLogo.svg";
import BackButton from "../../assets/images/BackButton.svg";
import "./Header.scss";

const Header = () => {
  return (
    <div className="header">
      <img src={BackButton} alt="" />
      <div className="header--container">
        <div className="header--logo-title">
          <img src={AppLogo} alt="" />
          <h1>LOCAL</h1>
        </div>
      </div>
    </div>
  );
};

export default Header;
