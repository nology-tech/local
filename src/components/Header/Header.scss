@use "../../styles/mixins" as *;

.header {
  display: flex;
  flex-direction: row;
  width: 100%;
  align-items: center;
  justify-content: space-between;
  padding: 0.4rem;
}
.header__container {
  display: flex;
  flex-direction: row;
  justify-content: center;
}

.header__logo-title {
  display: flex;
  flex-direction: row;
  margin: 2px;
}

.header__title-local {
  @include text-header;
}
