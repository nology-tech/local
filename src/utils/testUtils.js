import { jest } from "@jest/globals";
import { render } from "@testing-library/react";
import { BrowserRouter as Router } from "react-router-dom";
import UserContext from "../context/UserContext";
import mockUser from "../data/mockUser";

export const customRender = (
  ui,
  useRouting = true,
  contextOptions = {
    useContext: true,
    value: { user: mockUser, setUser: jest.fn() },
  }
) => {
  // wrap components in routing and context if requested
  const routingUiResult = useRouting && wrapWithRouting(ui);
  const contextUiResult =
    contextOptions.useContext &&
    wrapWithContext(routingUiResult, contextOptions.value);

  // use RTL's render function to return the test component
  return render(contextUiResult);
};

const wrapWithRouting = (ui) => {
  return <Router>{ui}</Router>;
};

const wrapWithContext = (ui, value) => {
  return <UserContext.Provider value={value}>{ui}</UserContext.Provider>;
};
