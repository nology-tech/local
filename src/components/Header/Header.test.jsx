import Header from "./Header";
import { customRender } from "../../utils/testUtils";

it("should render the header on the page", () => {
  const { container } = customRender(<Header />);
  expect(container).toMatchSnapshot();
});
