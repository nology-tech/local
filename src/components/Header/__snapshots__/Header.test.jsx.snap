// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the header on the page 1`] = `
<div>
  <div
    class="header"
  >
    <img
      alt=""
      src="BackButton.svg"
    />
    <div
      class="header__container"
    >
      <div
        class="header__logo-title"
      >
        <img
          alt=""
          src="AppLogo.svg"
        />
        <h1>
          LOCAL
        </h1>
      </div>
    </div>
  </div>
</div>
`;
